tests/utils.mjs

```diff
@@ -0,0 +1,168 @@
+import dns from 'dns';
+import https from 'https';
+import axios from 'axios';
+import { strict as assert } from 'node:assert'
+import { pipeline } from 'node:stream/promises'
+import { getPreset } from '../config/index.mjs'
+
+const dnsOptions = {
+  family: 4, // Don't use IPv6
+  all: true, // Return all addresses
+};
+
+// Create a global httpsAgent for requests with ignored SSL certificates
+const globalHttpsAgent = new https.Agent({ rejectUnauthorized: false });
+
+/**
+ * Helper method to resolve a hostname
+ */
+export async function resolveHost(host) {
+  let result;
+  try {
+    result = await dns.promises.lookup(host, dnsOptions);
+  } catch (err) {
+    return [false, `Failed to resolve host: ${host}`];
+  }
+  return [true, result.map((record) => record.address)];
+}
+
+export async function resolveHostAsIp(host) {
+  const result = (await resolveHost(host))[1];
+  return Array.isArray(result) && result.length > 0 ? result[0] : false;
+}
+
+/**
+ * Helper method to test a URL
+ */
+export async function testUrl(url, customOptions = {}) {
+  const options = {
+    method: 'GET',
+    headers: {},
+    data: undefined,
+    ignoreCertificate: true,
+    timeout: 1500,
+    returnAs: false,
+    returnError: false,
+    ...customOptions,
+  };
+
+  if (options.ignoreCertificate && url.trim().toLowerCase().startsWith('https://')) {
+    options.httpsAgent = globalHttpsAgent;
+  }
+
+  try {
+    const result = await axios(url, options);
+
+    switch (options.returnAs) {
+      case 'status':
+        return result.status;
+      case 'body':
+      case 'text':
+      case 'json':
+        return result.data;
+      case 'check':
+        return ![4, 5].includes(String(result.status).charAt(0));
+      default:
+        return result;
+    }
+  } catch (err) {
+    return options.returnError ? err : false;
+  }
+}
+
+export async function get(url, raw = false, log = false, ignoreCertificate = true) {
```

Comment:
I would call this `__withoutBody` (see below)

Reply:
I've changed the get function name to __withoutBody and give first parameter method("GET" or "DELETE") and modified function logic fit for both get and delete.